src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -45,6 +45,16 @@
  */
 class Serializer implements SerializerInterface, ContextAwareNormalizerInterface, ContextAwareDenormalizerInterface, ContextAwareEncoderInterface, ContextAwareDecoderInterface
 {
+    private const SCALAR_TYPES = [
+        'int' => true,
+        'integer' => true,
```

Comment:
this is not a native scalar type, no need to support it, same for boolean and double

Reply:
I don't think so because `gettype()` returns exactly this strings for int, bool and float values.

Reply:
I'm pretty sure we don't want to support types that PHP itself doesn't support.

Reply:
Probably I misunderstand you: why do you think that PHP doesn't support `integer`, `boolean` and `double` types? Or what do you mean under `doesn't support`?

When I wrote this code I was guided by the documentation for the `settype()`, in particular the description of `type` parameter and it possible values.

Reply:
```php
$f  = function (double $a) {};
$f(1.0);
```
> Uncaught TypeError: Argument 1 passed to {closure}() must be an instance of double, float given

Reply:
In terms of native type hinting you are right. But here we deal with type casting and do it by `settype()` that works well both with `int` and with `integer`.
Therefore, I find no reason to exclude support of `integer`, `boolean` and `double`.

Reply:
With this reasoning, PHP should have supported `function (double $a) {};`.
The `double` here is also an annotation - one that the engine understands.
I find it quite nice that PHP internal decide to provide one single alternative.
That helps consistency, everyone is forced to use one single syntax.
I wish we'd do the same and I see zero reason to support more.

Reply:
I have changed this, but I think that this thing should be noted in docs.